### ISSUE TITLE
Crashes when in RTL mode

### ISSUE THREAD

**Issue by elementaryBot** [elementary/files#6]
I can consistently produce a segmentation fault when using Files in RTL mode, steps to reproduce:
1. Open Files instance.
2. Switch to RTL mode (either in Gtk Inspector or using RTL language)
3. Switch the view type (e.g: from icon view to column view)
4. Segfault happens.

Here's the stacktrace:
#0  0x00007ffff6747860 in ?? () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
#1  0x00007ffff67481fb in ?? () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
#2  0x00007ffff6748214 in ?? () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
#3  0x00007ffff6748331 in ?? () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
#4  0x00007ffff59c8c4f in ?? () from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#5  0x00007ffff59c9d25 in g_object_unref () from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#6  0x00000000004564cc in fm_abstract_directory_view_finalize ()
#7  0x00007ffff59c9d9a in g_object_unref () from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#8  0x00007ffff6778057 in ?? () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
#9  0x00007ffff67bfab0 in ?? () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
#10 0x00007ffff59c4fa5 in g_closure_invoke () from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#11 0x00007ffff59d74de in ?? () from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#12 0x00007ffff59dfd5c in g_signal_emit_valist () from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#13 0x00007ffff59e008f in g_signal_emit () from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#14 0x00007ffff69c1b5a in ?? () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
#15 0x00007ffff59c9d25 in g_object_unref () from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#16 0x00007ffff4d5f199 in gof_abstract_slot_finalize ()
   from /usr/lib/x86_64-linux-gnu/libpantheon-files-core.so.0
#17 0x00007ffff59c9d9a in g_object_unref () from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#18 0x000000000045dbdc in marlin_view_view_container_change_view_mode ()
#19 0x0000000000456ad5 in _marlin_view_window_action_view_mode_gsimple_action_activate_callback ()
#20 0x00007ffff59c4fa5 in g_closure_invoke () from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#21 0x00007ffff59d6fc1 in ?? () from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#22 0x00007ffff59dfd5c in g_signal_emit_valist () from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#23 0x00007ffff59e0555 in g_signal_emit_by_name () from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#24 0x00007ffff5495b4b in __lambda15_ () from /usr/lib/x86_64-linux-gnu/libpantheon-files-widgets.so.0
#25 0x00007ffff5495bc5 in ___lambda15__granite_widgets_mode_button_mode_changed ()
   from /usr/lib/x86_64-linux-gnu/libpantheon-files-widgets.so.0
#26 0x00007ffff59c80a7 in g_cclosure_marshal_VOID__OBJECTv ()
   from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#27 0x00007ffff59c51d4 in ?? () from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#28 0x00007ffff59df9a6 in g_signal_emit_valist () from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#29 0x00007ffff59e0555 in g_signal_emit_by_name () from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#30 0x00007ffff7b55267 in granite_widgets_mode_button_set_active ()
   from /usr/lib/x86_64-linux-gnu/libgranite.so.3
#31 0x00007ffff7b56267 in granite_widgets_mode_button_set_selected ()
   from /usr/lib/x86_64-linux-gnu/libgranite.so.3
#32 0x00007ffff7b54707 in __lambda23_ () from /usr/lib/x86_64-linux-gnu/libgranite.so.3
#33 0x00007ffff7b54726 in ___lambda23__gtk_toggle_button_toggled ()
   from /usr/lib/x86_64-linux-gnu/libgranite.so.3

Launchpad Details: [#LP1695743](https://bugs.launchpad.net/bugs/1695743) Adam Bieńkowski - 2017-06-04 14:24:41 +0000

**Comment by jeremypw** [comment]
I have confirmed this using Gtk Inspector (GTK_DEBUG=interactive pantheon-files), however under those conditions I also got a crash in LTR mode, so it does not appear to be related to RTL.

I also tried LANGUAGE=ar_DZ.UTF.8 pantheon-files and under these conditions I did not get a crash changing view mode.

I am not sure this is a bug in Files therefore, or an artifact of running under Gtk.Inspector.  The crash occurs during the destruction of     at /media/jeremy/Shared/shared_data/Vala/Elementary/pantheon-files-branches/files/src/View/AbstractDirectoryView.vala
